src/types/api.type.ts

```diff
@@ -0,0 +1,21 @@
+import { HTTPMethods } from './common.types';
+
+export interface ApiRequestParams {
+  url: string;
+  method: HTTPMethods;
+  baseURL?: string;
+  headers?: Record<string, string>;
+  data?: unknown;
+  [key: string]: any;
+}
+
+export interface ApiResponse<T = any> {
+  data: T;
+  status: number;
+  statusText: string;
+  headers: Record<string, string>;
+  config: ApiRequestParams;
```

Comment:
Why do we need config?

Reply:
In response we are sending config back as response. Basically to debug. I can remove this. It's an additional info. If needed?

Reply:
removed